### PR TITLE
Add a files option, which loads a key-value mapping from .json files

### PR DESCRIPTION
**Pull request by jafl**
Built on top of PR #3, this adds a file option, which loads a key-value mapping from the specified .json file.

I couldn't find any other way to provide actual string data to the template processing.  Is there a better way?

### CODE REVIEW

**Comment by jafl** [comment]
There is another way to do this:
```{ options: Object.assign(require(my_json_file), {...extra options...}) }```

However, I still think a files option is clearer.

File: index.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
-var cons = require('consolidate');
-var utils = require('loader-utils');
-var extname = require('path').extname;
+const cons = require('consolidate'),
+      utils = require('loader-utils'),
+      path = require('path'),
+      fs = require('fs');
```
Comment (jtangelder) [inline index.js]: could you undo this formatting?

File: index.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
 
 module.exports = function(content) {
   this.cacheable && this.cacheable();
 
-  var callback = this.async();
-  var opt = utils.parseQuery(this.query);
+  const callback = this.async();
```
Comment (jtangelder) [inline index.js]: leave this to var, unrelated code change.

File: index.js [hunk 3]
```diff
@@ -28,6 +29,20 @@ module.exports = function(content) {
...
+  opt.dirname = path.dirname(this.resourcePath);
+
+  const self = this;
+  if(opt.files instanceof Array) {
+    opt.files.reverse().forEach(function(file) {
+      var file = path.join(opt.dirname, file + '.json');
+      if(!fs.existsSync(file)) {
+        throw new Error("Data file '"+ file +"' does not exist");
+      }
+      opt = Object.assign(JSON.parse(fs.readFileSync(file)), opt);    // ensure that opt takes precedence
```
Comment (jtangelder) [inline index.js]: code comment above the rule please.

**Review by jtangelder — CHANGES_REQUESTED** [review]
Thanks, some small comments and then it is good to go!

**Comment by jafl** [comment]
I'm so sorry this took so long!  I never received the email from github. with your feedback.

**Comment by jafl** [comment]
Just pinging to see if it can be merged.

**Comment by jtangelder** [comment]
Thank you John, LGTM.

**Pull request merged**